Code/Framework/AzFramework/Platform/Android/AzFramework/Components/NativeUISystemComponentFactories_Android.h

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#pragma once
+
+#include <AzFramework/Application/Application.h>
+#include <AzFramework/Input/Devices/Keyboard/InputDeviceKeyboard.h>
+#include <AzFramework/Input/Devices/Motion/InputDeviceMotion.h>
+#include <AzFramework/Input/Devices/Mouse/InputDeviceMouse.h>
+#include <AzFramework/Input/Devices/Touch/InputDeviceTouch.h>
+#include <AzFramework/Input/Devices/VirtualKeyboard/InputDeviceVirtualKeyboard.h>
+#include <AzFramework/Windowing/NativeWindow.h>
+
+////////////////////////////////////////////////////////////////////////////////////////////////////
+namespace AzFramework
+{
+    class AndroidApplicationImplFactory 
+        : public Application::ImplementationFactory
+    {
+    public:
+        AZStd::unique_ptr<Application::Implementation> Create() override;
+    };
+
+    class AndroidDeviceKeyboardImplFactory
+        : public InputDeviceKeyboard::ImplementationFactory
+    {
+    public:
+        AZStd::unique_ptr<InputDeviceKeyboard::Implementation> Create(InputDeviceKeyboard& inputDevice) override;
+    };
+
+    class AndroidDeviceMotionImplFactory
+        : public InputDeviceMotion::ImplementationFactory
+    {
+    public:
+        AZStd::unique_ptr<InputDeviceMotion::Implementation> Create(InputDeviceMotion& inputDevice) override;
+    };
+
+    class AndroidDeviceMouseImplFactory
+        : public InputDeviceMouse::ImplementationFactory
+    {
+    public:
+        AZStd::unique_ptr<InputDeviceMouse::Implementation> Create(InputDeviceMouse& inputDevice) override;
+    };
+
+    class AndroidDeviceTouchImplFactory
+        : public InputDeviceTouch::ImplementationFactory
+    {
+    public:
+        AZStd::unique_ptr<InputDeviceTouch::Implementation> Create(InputDeviceTouch& inputDevice) override;
+    };
+
+    class AndroidDeviceVirtualKeyboardImplFactory
+        : public InputDeviceVirtualKeyboard::ImplementationFactory
+    {
+    public:
+        AZStd::unique_ptr<InputDeviceVirtualKeyboard::Implementation> Create(InputDeviceVirtualKeyboard& inputDevice) override;
+    };
+
+    class AndroidNativeWindowFactory 
+        : public NativeWindow::ImplementationFactory
+    {
+    public:
+        AZStd::unique_ptr<NativeWindow::Implementation> Create() override;
+    };
```

Comment:
They were placed in a header file on purpose because the implementation of the ::Create functions are in the pimpl class functions. 
For example in [InputDeviceKeyboard_Android](https://github.com/aws-lumberyard-dev/o3de/blob/f4212b1bf9296f08b7ae59c996d11dd3c2171cf9/Code/Framework/AzFramework/Platform/Android/AzFramework/Input/Devices/Keyboard/InputDeviceKeyboard_Android.cpp#L438-L441)

```
 AZStd::unique_ptr<InputDeviceKeyboard::Implementation> AndroidDeviceKeyboardImplFactory::Create(InputDeviceKeyboard& inputDevice)
    {
        return AZStd::make_unique<InputDeviceKeyboardAndroid>(inputDevice);
    }
```

Reply:
But for Linux and Windows, you placed the implementation in their cpp files, so it is inconsistent with those platforms.

Reply:
I was able to do it for Linux because there was another level of abstraction using Xcb, so most of the device specific implementation was wrapped in other headers anyways. For windows, it was inverted where the pimpl classes had header files and was included in the component class. 
I could remove the header files if i move the PAL platform implementation of in the pimpl classes, but that would mean i need to keep the base ones for non-supported devices

Reply:
Alright this is fine then, I am not going to block the PR over moving the Android, MacOS and iOS code out of the header.

Reply:
Its not a bad idea, i'm going to make that pattern consistent